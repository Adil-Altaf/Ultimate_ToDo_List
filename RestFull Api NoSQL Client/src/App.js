import React, { Component } from 'react';
import  {Provider} from 'react-redux';
import store from './store/index';
import Home from './components/home/Home';
import Task from './components/task/Task';
import {BrowserRouter as  Router  , Route ,  } from 'react-router-dom';
class App extends Component {


  render() {
    return (
      <Provider store = {store} >
      <div>
        <Router>
          <div>
          <Route exact path = "/" component = {Home} />
          <Route exact path = "/tasks" component = {Task} />
          </div>
        </Router>
        </div>
      </Provider>
    );
  }
}

export default App;

