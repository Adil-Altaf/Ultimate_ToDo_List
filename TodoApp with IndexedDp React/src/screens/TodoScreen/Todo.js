import React, { Component } from 'react';
import '../../Style/todo.css';
import db from './dexie';
import TodoList from './TodoList';

class Todo extends Component {
    //Initlize The Component State 
    constructor() {
        super();
        this.state = {
            title: '',
            description: '',
            todos: [],
            exist: false
        }
        this.submitForm = this.submitForm.bind(this);
        this.edit = this.edit.bind(this);
        this.deleteList = this.deleteList.bind(this);
    }
    //To get all Todos From IndexedDb
    componentDidMount() {
        const { exist } = this.state;
        db.table('todos')
            .toArray()
            .then((todos) => {

                console.log(todos.length);

                console.log(exist);

                if (todos.length === 1) {
                    if (todos[0].todo.length > 1 || todos.length > 0) {
                        this.setState({ exist: true })
                        this.setState({ todos: todos[0].todo });
                    }
                }

            });
    }


    //To Handle Input
    updateTitle(title) {
        this.setState({ title: title })
    }
    //To Handle Input
    updateDescription(description) {
        this.setState({ description: description });
    }
    //To Submit Form
    submitForm(e) {
        e.preventDefault();
        const { title, description, todos, exist } = this.state;
        console.log(exist);

        todos.push({ title, description, completed: false });
        let todo = todos
        if (!exist) {
            db.table('todos')
                .add({ id: 1, todo })
                .then((id) => {
                    console.log(id);
                    this.setState({ exist: true })
                });
        } else {
            db.table('todos')
                .update(1, { todo })
                .then((x) => {
                    console.log(x);
                });
        }


        this.setState({ todos, title: '', description: '' });
    }
    //To delete a specific todo from state and from indexed Db
    deleteList(index) {
        const { todos } = this.state;
        let updated = todos.filter((item, index1) => {
            if (index1 !== index) {
                return true
            }
            else {
                return false
            }

        });
        this.setState({ todos: updated })
        let todo = updated;
        console.log(updated);
        db.table('todos')
            .update(1, { todo })
            .then((x) => {
                console.log(x);
            });


    }
    //To Edit a specific todo
    edit(index) {
        const { todos } = this.state;
        todos[index].completed = !todos[index].completed;
        this.setState(todos);
        let todo = todos;
        console.log(todo);
        db.table('todos')
            .update(1, { todo })
            .then((x) => {
                console.log(x);
            });
    }
    //Input Field for title and description
    inputField() {
        const { title, description } = this.state;
        return (
            <form className="form" onSubmit={this.submitForm} >
                <div>
                    <label style={{ marginRight: 2 }}>
                        Title
                        <input
                            required
                            value={title}
                            onChange={(e) => this.updateTitle(e.target.value)}
                            type="text"
                            className="form-control"
                        />
                    </label>
                    <div>
                        <label style={{ marginLeft: 2 }} >
                            Description
                        <textarea
                                required
                                value={description}
                                onChange={(e) => this.updateDescription(e.target.value)}
                                type="text"
                                className="form-control"
                            />
                        </label>
                    </div>
                </div>
                <button className="btn btn-primary">
                    Add Todo
                </button>
            </form>
        )
    }

    render() {
        const { todos } = this.state;
        return (
            <div style={{ textAlign: "center" }} >
                <h1>
                    TodoApp
                </h1>
                {this.inputField()}
                {/* Breaking The Todos in Component */}
                <TodoList todos={todos} edit={this.edit} deleteList={this.deleteList} />
            </div>
        )
    }
}
export default Todo;