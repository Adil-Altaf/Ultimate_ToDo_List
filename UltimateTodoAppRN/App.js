import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import reducers from "./src/store/reducers/index";
import thunk from "redux-thunk";
import TodoList from "./src/TodoList";

const store = createStore(reducers, {}, applyMiddleware(thunk));

export default class App extends React.Component {

  render() {
    return (
      <Provider store={store}>
        <TodoList />
      </Provider>
    );
  }
}
