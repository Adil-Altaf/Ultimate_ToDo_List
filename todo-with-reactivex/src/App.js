import React, { Component } from 'react';
import Home from './components/home/Home';
import Task from './components/task/Task';
import {BrowserRouter as  Router  , Route ,  } from 'react-router-dom';
class App extends Component {


  render() {
    
    return (
      <div>
        <Router>
          <div>
          <Route exact path = "/" component = {Home} />
          <Route exact path = "/tasks" component = {Task} />
          </div>
        </Router>
        </div>
    )  
}
}

export default App;

