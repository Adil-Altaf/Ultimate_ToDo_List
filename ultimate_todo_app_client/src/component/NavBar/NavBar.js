import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';



const NavBar = (props) => {
    return (
        <div>
            <AppBar position="static" style={{backgroundColor:'purple',textAlign:'center'}} >
                <Toolbar style={{textAlign:'center'}}>
                    <Typography variant="title" color="inherit" >
                        {props.children}
                    </Typography>
                </Toolbar>
            </AppBar>
            
        </div>
    )
}

export default NavBar;