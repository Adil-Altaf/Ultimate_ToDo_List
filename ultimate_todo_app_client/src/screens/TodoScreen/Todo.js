import React, { Component } from 'react';
import '../../Style/todo.css';

class Todo extends Component {

    //Initlize The Component State
    constructor() {
        super();
        this.state = {
            title: '',
            description: '',
            todos: []
        }
        this.submitForm = this.submitForm.bind(this);
    }
    //To Handle Input
    updateTitle(title) {
        this.setState({ title: title })
    }
    //To Handle Input
    updateDescription(description) {
        this.setState({ description: description });
    }
    //To Submit Form
    submitForm(e) {
        e.preventDefault();
        const { title, description, todos } = this.state;
        todos.push({ title, description, completed: false });
        this.setState({ todos, title: '', description: '' });
    }
    //To delete a specific todo
    deleteList(index) {
        const { todos } = this.state;
        let updated = todos.filter((item, index1) => {
            if (index1 !== index) {
                return true
            }
            else{
                return false
            }

        });
        this.setState({ todos: updated })
        console.log(updated);


    }
    //To Edit a specific todo
    edit(index) {
        const { todos } = this.state;
        todos[index].completed = !todos[index].completed;
        this.setState(todos);
    }
    //input field for title and description
    inputField() {
        const { title, description } = this.state;
        return (//Creating from of Title And Description
            <form className="form" onSubmit={this.submitForm} >
                <div>
                    <label style={{ marginRight: 2 }}>
                        Title
                        <input
                            required
                            value={title}
                            onChange={(e) => this.updateTitle(e.target.value)}
                            type="text"
                            className="form-control"
                        />
                    </label>
                    <div>
                        <label style={{ marginLeft: 2 }} >
                            Description
                        <textarea
                                required
                                value={description}
                                onChange={(e) => this.updateDescription(e.target.value)}
                                type="text"
                                className="form-control"
                            />
                        </label>
                    </div>
                </div>
                <button className="btn btn-primary">
                    Add Todo
                </button>
            </form>
        )
    }
    unorderList() {
        const { todos } = this.state;
        return (//Todo List 
            <div>

                <ul className='listStyle'>
                    {todos.map((list, index) => {
                        return (
                            <li key={index + list.title}>
                                {list.completed ?
                                    <span className='completed'>
                                        {list.title}-{list.description}
                                    </span>

                                    :

                                    <span className=''>
                                        {list.title}-{list.description}
                                    </span>


                                }
                                <span className="delete" >
                                    {list.completed ?
                                        <i className="fa fa-check-square-o actionButton edit1" onClick={() => this.edit(index)}></i>

                                        :

                                        <i className="fa fa-square-o actionButton edit1" onClick={() => this.edit(index)}></i>
                                    }

                                    <i className="fa fa-remove actionButton delete1" onClick={() => this.deleteList(index)} ></i>
                                </span>
                                <hr />
                            </li>
                        )
                    })}
                </ul>
            </div>
        )
    }



    render() {
        return (
            <div style={{ textAlign: "center" }} >
                <h1>
                    TodoApp
                </h1>
                {this.inputField()}
                {this.unorderList()}
            </div>
        )
    }
}
export default Todo;