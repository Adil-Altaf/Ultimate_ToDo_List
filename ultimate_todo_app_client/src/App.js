import React, { Component } from 'react';
import Todo from './screens/TodoScreen/Todo';
import './Style/bootstrap.min.css';
import '../node_modules/font-awesome/css/font-awesome.min.css'
import NavBar from './component/NavBar/NavBar';

class App extends Component {
  render() {
    return (
      <div className="App">
        <NavBar>
          Ultimate Todo App
        </NavBar>
        <Todo />
      </div>
    );
  }
}

export default App;
