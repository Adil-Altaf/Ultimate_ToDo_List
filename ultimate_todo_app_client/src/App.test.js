import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Todo from './screens/TodoScreen/Todo';
import NavBar from './component/NavBar/NavBar';
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders Todo List  without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Todo />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Header Should Work Fine', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NavBar>Some Text</NavBar>, div);
  ReactDOM.unmountComponentAtNode(div);
});