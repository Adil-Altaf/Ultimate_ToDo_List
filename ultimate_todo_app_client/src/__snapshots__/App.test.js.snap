// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="App"
>
  <div>
    <header
      className="MuiPaper-root-075 MuiPaper-elevation4-081 MuiAppBar-root-066 MuiAppBar-positionStatic-070 MuiAppBar-colorPrimary-073"
      style={
        Object {
          "backgroundColor": "purple",
        }
      }
    >
      <div
        className="MuiToolbar-root-0102 MuiToolbar-regular-0104 MuiToolbar-gutters-0103"
        style={
          Object {
            "margin": "auto",
            "textAlign": "center",
          }
        }
      >
        <h2
          className="MuiTypography-root-0106 MuiTypography-title-0112 MuiTypography-colorInherit-0125"
        >
          Ultimate Todo App
        </h2>
      </div>
    </header>
  </div>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h1>
      TodoApp
    </h1>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <div>
        <label
          style={
            Object {
              "marginRight": 2,
            }
          }
        >
          Title
          <input
            className="form-control"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
        </label>
        <div>
          <label
            style={
              Object {
                "marginLeft": 2,
              }
            }
          >
            Description
            <textarea
              className="form-control"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </label>
        </div>
      </div>
      <button
        className="btn btn-primary"
      >
        Add Todo
      </button>
    </form>
    <div>
      <ul
        className="listStyle"
      />
    </div>
  </div>
</div>
`;

exports[`NavBar also has a valid snapshot 1`] = `
<div>
  <header
    className="MuiPaper-root-075 MuiPaper-elevation4-081 MuiAppBar-root-066 MuiAppBar-positionStatic-070 MuiAppBar-colorPrimary-073"
    style={
      Object {
        "backgroundColor": "purple",
      }
    }
  >
    <div
      className="MuiToolbar-root-0102 MuiToolbar-regular-0104 MuiToolbar-gutters-0103"
      style={
        Object {
          "margin": "auto",
          "textAlign": "center",
        }
      }
    >
      <h2
        className="MuiTypography-root-0106 MuiTypography-title-0112 MuiTypography-colorInherit-0125"
      />
    </div>
  </header>
</div>
`;

exports[`Todos has a valid snapshot 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    TodoApp
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div>
      <label
        style={
          Object {
            "marginRight": 2,
          }
        }
      >
        Title
        <input
          className="form-control"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </label>
      <div>
        <label
          style={
            Object {
              "marginLeft": 2,
            }
          }
        >
          Description
          <textarea
            className="form-control"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
        </label>
      </div>
    </div>
    <button
      className="btn btn-primary"
    >
      Add Todo
    </button>
  </form>
  <div>
    <ul
      className="listStyle"
    />
  </div>
</div>
`;
